plugin/trino-clickhouse/src/main/java/io/trino/plugin/clickhouse/ClickHouseClient.java

```diff
@@ -271,6 +273,39 @@ public boolean supportsAggregationPushdown(ConnectorSession session, JdbcTableHa
         return preventTextualTypeAggregationPushdown(groupingSets);
     }
 
+    @Override
+    public boolean supportsTopN(ConnectorSession session, JdbcTableHandle handle, List<JdbcSortItem> sortOrder)
+    {
+        for (JdbcSortItem sortItem : sortOrder) {
+            Type sortItemType = sortItem.column().getColumnType();
+            if (sortItemType instanceof CharType || sortItemType instanceof VarcharType) {
```

Comment:
When do we receive char type in the connector?

Reply:
As far as I can see, clickhouse-connector doesn't specifically map anything to CharType, preferring VarcharType instead. However, the check like `x instanceof CharType || x instanceof VarcharType` is already present in the connector (see `toWriteMapping`), therefore I'd like to be consistent with existing implementation. Plus it's a common check for other connectors.

Please let me know if I shall drop charType?

Reply:
A char type in toWriteMapping is reachable. e.g. `CREATE TABLE test (a char(1))`. I suppose other connectors support the type in toColumnMapping method. 

I would recommend throwing an exception instead of dropping if getting char types in this loop is unexpected.

Reply:
Changed. Exception will be thrown upon encountering `CharType`, also added a simple unit test to make sure TopN(varchar) is not pushed down.

I don't think that this change adds a value though. Previous implementation would just not apply TopN pushdown optimization, which is the desired behavior.